src/main/java/com/ocient/jdbc/XGConnection.java

```diff
@@ -1455,6 +1455,25 @@ private void saveSecondaryInterfaces(final List<String> newCmdComps, final List<
 		}		
 	}
 
+	private CompletionStage<Void> clientHandshakeAsync(final String userid, final String pwd, final String db, final boolean shouldRequestVersion) {
+		CompletionStage<Void> clientHandshakeCompletionStage = CompletableFuture.runAsync(() -> {
+			try { clientHandshake(user, pwd, database, shouldRequestVersion); }
+			catch (Exception e) {
+				LOGGER.log(Level.WARNING, String.format("Handshake exception %s occurred in reconnect() with message %s", e.toString()));
+			}
+		});
+
+		return clientHandshakeCompletionStage.thenRun(() -> {
+			try {
+				clientHandshakeFinalize();
+			} catch (Exception e) {
+				LOGGER.log(Level.WARNING, String.format("Handshake exception %s occurred in reconnect() with message %s", e.toString()));
+			}
+		});
```

Comment:
this method now returns a future, so you'd want to "chain" _this_ stage onto the completion of the previous stage `clientHandshakeCompletionStage` using composition, i.e. `thenCompose(...)`. 
```suggestion
		return clientHandshakeCompletionStage.thenCompose(this::clientHandshakeFinalize);
```